db/upgrade.php

```diff
@@ -628,5 +628,32 @@ function xmldb_attendance_upgrade($oldversion=0) {
 
     }
 
+    if ($oldversion < 2019080802) {
+
+        // Changing nullability of field sessiondetailspos on table attendance to not null.
+        $table = new xmldb_table('attendance');
+        $field = new xmldb_field('sessiondetailspos', XMLDB_TYPE_CHAR, '5', null, XMLDB_NOTNULL, null, 'left', 'subnet');
+
+        // Launch change of nullability for field sessiondetailspos.
+        $dbman->change_field_notnull($table, $field);
+
+        // Changing the default of field preventsharedip on table attendance_sessions to 1.
+        $table = new xmldb_table('attendance_sessions');
+        $field = new xmldb_field('preventsharedip', XMLDB_TYPE_INTEGER, '1', null, XMLDB_NOTNULL, null, '1', 'absenteereport');
```

Comment:
the install.xml file states a default of 0 for preventsharedip - sounds like you have a weird install.xml file?